### PR TITLE
Update Sync Serverless Application 

### PR DESCRIPTION
**Pull request by manodnyab**
## Description

1) Update Sync Serverless Application to pull in the new changes made by SAM CLI
2) Remove Sync Serverless Application from Experimental
3) Remove the Experimental Sync Serverless Application Code Only option

## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/sam/SamExecutable.kt [hunk 1]
```diff
@@ -305,7 +304,6 @@ fun GeneralCommandLine.samSyncCommand(
         addParameter("--use-container")
     }
     addParameter("--no-dependency-layer")
-    if (syncOnlyCode) {
-        addParameter("--code")
-    }
+
+    addParameter("--no-watch")
```
Comment (rli) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/sam/SamExecutable.kt:308]: do we need to worry about older SAM CLI versions that can't handle this flag?

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/actions/SyncServerlessAppAction.kt [hunk 2]
```diff
@@ -75,8 +75,8 @@ class SyncServerlessAppAction(private val codeOnly: Boolean = false) : AnAction(
             }
 
             val execVersion = SemVer.parseFromText(samExecutable.version) ?: error("SAM CLI version could not detected")
-            val minVersion = SemVer("1.53.0", 1, 53, 0)
-            val minVersionForUseContainer = SemVer("1.57.0", 1, 57, 0)
+            val minVersion = SemVer("1.78.0", 1, 78, 0)
```
Comment (rli) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/actions/SyncServerlessAppAction.kt:78]: maybe we should use the breaking change changelog entry

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/sam/SamExecutable.kt (hunk 1 above)
Comment (manodnyab) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/sam/SamExecutable.kt:308]: This new change requires an upgrade to the latest version that supports this change aka 1.78.0

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/actions/SyncServerlessAppAction.kt (hunk 2 above)
Comment (manodnyab) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/actions/SyncServerlessAppAction.kt:78]: Added changelog

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/actions/SyncServerlessAppAction.kt [hunk 3]
```diff
@@ -108,15 +108,11 @@ class SyncServerlessAppAction(private val codeOnly: Boolean = false) : AnAction(
...
-                            } catch (e: Exception) {
-                                true
-                            }
-                        } else {
-                            null
+                        val dockerDoesntExist = try {
+                            val processOutput = ExecUtil.execAndGetOutput(GeneralCommandLine("docker", "ps"))
+                            processOutput.exitCode != 0
+                        } catch (e: Exception) {
+                            true
```
Comment (rli) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/actions/SyncServerlessAppAction.kt:115]: probably log error

**Review by rli — APPROVED** [review]

File: resources/resources/software/aws/toolkits/resources/MessagesBundle.properties [hunk 4]
```diff
@@ -1225,7 +1225,7 @@ serverless.application.deploy.validation.template.values.tooLong=Template value
 serverless.application.deploy.validation.template.values.tooShort=Template value for {0} does not meet MinLength defined in template: {1}
 serverless.application.deploy.validation.template.values.tooSmall=Template value for {0} is smaller than MinValue defined in template: {1}
 serverless.application.deploy_in_progress.title=Deploying Application {0}
-serverless.application.sync=Sync Serverless Application
+serverless.application.sync=Sync Serverless Application(formerly Deploy)
```
Comment (rli) [inline resources/resources/software/aws/toolkits/resources/MessagesBundle.properties:1228]:
```suggestion
serverless.application.sync=Sync Serverless Application (formerly Deploy)
```

**Pull request merged**